### PR TITLE
[FLINK-26100][docs] Set up Flink ML Document Website

### PR DESCRIPTION
**Pull request by yunfengzhou-hub**
## What is the purpose of the change

- Set up Flink ML Document Website

## Brief change log

- Add `docs` folder containing all configurations needed to setup the framework for Flink ML's document website.
- Add basic documentation about example operator, quick start and release announcement blog.
- Add guidelines to build and start the website, as well as to contribute documents to `docs` folder in README.

## Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): (no)
- The public API, i.e., is any changed class annotated with @public(Evolving): (no)
- Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: docs/content/posts/release-ml-2.0.0.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+The Apache Flink community is excited to announce the release of Flink ML
```
Comment (lindong28) [inline docs/content/posts/release-ml-2.0.0.md:35]: Could we avoid copy/paste the entire blog post from flink-web? We can use URL if needed.

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR! LGTM overall. Left one minor comment.

File: docs/content/posts/release-ml-2.0.0.md (hunk 1 above)
Comment (yunfengzhou-hub) [reply docs/content/posts/release-ml-2.0.0.md]: According to offline discussions, I would remove the blog post section from this PR. Blogs about Flink ML can be placed directly in flink-web repo.

**Comment by yunfengzhou-hub** [comment]
Thanks for the comments. I have fixed the issue mentioned in the comments above. We still need to deploy the contents of this PR to Flink document's website, but since this PR has already got a lot to review, I think it could be better to leave the deployment configuration and script changes to the next PR.

**Comment by lindong28** [comment]
Thanks for the update. LGTM.

**Comment by lindong28** [comment]
Here are some links that related to setting up the Flink ML website and determine its build health:
-  Wiki for setting up Flink nightly website [link](https://cwiki.apache.org/confluence/display/FLINK/Managing+Documentation)
- Repo to create/update the scripts which pull docs from Flink ML [link](https://github.com/apache/infrastructure-bb2)
- Flink ML website build job history [link](https://ci2.apache.org/#/builders/103)

**Pull request merged**